StandaloneMmPkg/Library/StandaloneMmCoreEntryPoint/Arm/StandaloneMmCoreEntryPoint.c

```diff
@@ -40,89 +42,394 @@
 
 #include <Protocol/PiMmCpuDriverEp.h>
 
-#define SPM_MAJOR_VER_MASK   0xFFFF0000
-#define SPM_MINOR_VER_MASK   0x0000FFFF
-#define SPM_MAJOR_VER_SHIFT  16
-#define FFA_NOT_SUPPORTED    -1
-
-#define BOOT_PAYLOAD_VERSION  1
-
 extern EFI_MM_SYSTEM_TABLE  gMmCoreMmst;
 
-STATIC CONST UINT32  mSpmMajorVer = SPM_MAJOR_VERSION;
-STATIC CONST UINT32  mSpmMinorVer = SPM_MINOR_VERSION;
-
-STATIC CONST UINT32  mSpmMajorVerFfa = SPM_MAJOR_VERSION_FFA;
-STATIC CONST UINT32  mSpmMinorVerFfa = SPM_MINOR_VERSION_FFA;
+VOID  *gHobList = NULL;
 
 STATIC PI_MM_CPU_DRIVER_ENTRYPOINT  CpuDriverEntryPoint = NULL;
+STATIC MP_INFORMATION_HOB_DATA      *mMpInfo            = NULL;
 
 /**
-  Retrieve a pointer to and print the boot information passed by privileged
-  secure firmware.
+  Get ABI protocol.
 
-  @param  [in] SharedBufAddress   The pointer memory shared with privileged
-                                  firmware.
+  @retval         AbiProtocolSpmMm         SPM_MM ABI
+  @retval         AbiProtocolFfa           FF-A ABI
+  @retval         AbiProtocolUnknown       Invalid ABI.
 
 **/
-EFI_SECURE_PARTITION_BOOT_INFO *
-GetAndPrintBootinformation (
-  IN VOID  *SharedBufAddress
+STATIC
+ABI_PROTOCOL
+EFIAPI
+GetAbiProtocol (
+  IN VOID
   )
 {
-  EFI_SECURE_PARTITION_BOOT_INFO  *PayloadBootInfo;
-  EFI_SECURE_PARTITION_CPU_INFO   *PayloadCpuInfo;
-  UINTN                           Index;
+  UINT16        RequestMajorVersion;
+  UINT16        RequestMinorVersion;
+  UINT16        CurrentMajorVersion;
+  UINT16        CurrentMinorVersion;
+  ARM_SVC_ARGS  SvcArgs;
+  ABI_PROTOCOL  AbiProtocol;
+
+  ZeroMem (&SvcArgs, sizeof (ARM_SVC_ARGS));
+  SvcArgs.Arg0 = ARM_SVC_ID_FFA_VERSION_AARCH32;
+  SvcArgs.Arg1 = ((SPM_MAJOR_VERSION_FFA <<  16) | (SPM_MINOR_VERSION_FFA));

```

Comment:
This would be removed by https://github.com/tianocore/edk2/pull/6116/commits/525a0b18072c3530af90076f871ba67437b09936#diff-84a9f255c38cc05cb17b932227609c2673dc1435320d6f7f4db9193050824f31L76
After ArmFfaSvc.h is modified.